### PR TITLE
See also `axes` from `eachindex`'s docstring, etc.

### PR DESCRIPTION
**Pull request by mcabbott**
As suggested here: https://discourse.julialang.org/t/eachindex-for-individual-axes-of-multidimensional-arrays/81264/5

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -308,17 +311,21 @@ if all inputs have fast linear indexing, a [`CartesianIndices`](@ref)
 otherwise).
 If the arrays have different sizes and/or dimensionalities, a `DimensionMismatch` exception
```
[the remark this answers is not in the document]
Comment (c42f) [reply base/abstractarray.jl:312]: In English "or" often means `xor`, so and/or disambiguates that case. It's a bit awkward.
https://en.wikipedia.org/wiki/And/or

**Review by c42f — APPROVED** [review]
Looks good to me

File: base/abstractarray.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcabbott) [reply base/abstractarray.jl:312]: But that seems to make it worse: If differing dimensionalities means `ndims`, that implies differing `size`.

In my examples, with differing `ndims`, some quite different sizes are permitted (`(3,) and (1, 3)`) while other quite similar sizes are not (`(3,) and (3, 1)`).

[the remark this answers is not in the document]
Comment (mcabbott) [reply base/abstractarray.jl:312]: Ok, although it still seems an odd choice to describe computer logic, not a menu. And if differing dimensionalities means `ndims`, that implies differing `size` anyway.

While differing `size` might be necessary for an error, it's not sufficient. Can we describe the actual behaviour accurately? Or are some of those bugs?

**Comment by mcabbott** [comment]
Can this be merged?

**Comment by mcabbott** [comment]
Bump?

**Pull request merged**